### PR TITLE
Reject tuple index for missing fields assist

### PR DESCRIPTION
**Pull request by edwin0cheng**
Note that although we could add a diagnostics to fill the missing tuple field, e.g :

```
struct A(usize);
let a = A { 0: () }
```

But IMHO it is uncommon usage and it should not be encouraged.

### CODE REVIEW

**Comment by matklad** [comment]
Agree that we should skip the fix, but we *should* emit the diagnostic itself

**Comment by edwin0cheng** [comment]
Fixed 👍

File: crates/ra_ide/src/diagnostics.rs [hunk 1]
```diff
@@ -66,6 +66,14 @@ pub(crate) fn diagnostics(db: &RootDatabase, file_id: FileId) -> Vec<Diagnostic>
     .on::<hir::diagnostics::MissingFields, _>(|d| {
         let mut field_list = d.ast(db);
         for f in d.missed_fields.iter() {
+            // Note that although we could add a diagnostics to
+            // fill the missing tuple field, e.g :
+            // `struct A(usize);`
+            // `let a = A { 0: () }`
+            // but it is uncommon usage and it should not be encouraged.
+            if f.as_tuple_index().is_some() {
+                continue;
```
Comment (matklad) [inline crates/ra_ide/src/diagnostics.rs]: This still will create the fix. 

Maybe

```rust
let fix = if d.missed_fields.iter().any(|it| it.as_tuple_index().is_some()) { None } else { Some(...)}
```

Comment (edwin0cheng) [reply crates/ra_ide/src/diagnostics.rs]: Oh, I got what you mean now. 🤦‍♂

**Comment by edwin0cheng** [comment]
bors hate me 😞

**Comment by matklad** [comment]
No, bors hates when I edit github actions. I have a theory that *rebasing*
on top of master can help in similar situations

**Comment by edwin0cheng** [comment]
OK, if this happens again, I will try to do this. And I believe it will happen soon 😭

**Pull request merged**